### PR TITLE
QR reader supports recovery and connect code

### PR DESCRIPTION
**Pull request by cmonfortep**
Task/Issue URL: https://app.asana.com/0/488551667048375/1205212671337474/f 

### Description
Sync QR code reader supports both recovery and connect code, no matter which auth flow the user started.

### Steps to test this PR
(for testing you will need 2 devices)

_Feature 1_
- [x] install app in 2 devices, both unauthenticated
- [x] on deviceA, enable Sync
- [x] on deviceB, go to settings -> sync -> toggle On
- [x] Click on Turn On sync
- [x] Click on Sync another device
- [x] Read QR code from device A
- [x] Ensure device B logs into device A account
- [ ] ...
- [x] on device B, logout from sync account (toggle off)
- [x] on device B, toggle on again
- [x] Click on "Recover your synced data"
- [x] read QR code from device A
- [x] Ensure device B logs into device A account

### UI changes
| Before  | After |
| ------ | ----- |
!(Upload before screenshot)|(Upload after screenshot)|

### CODE REVIEW

**Comment by cmonfortep** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3486** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3486" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3486?utm_source=stack-comment).

File: sync/sync-impl/src/test/java/com/duckduckgo/sync/impl/AppSyncAccountRepositoryTest.kt [hunk 1]
```diff
@@ -359,19 +359,19 @@ class AppSyncAccountRepositoryTest {
     }
 
     @Test
-    fun whenConnectingUsingQRFromAuthenticatedDeviceThenConnectsDevice() {
+    fun whenConnectingFromAuthenticatedDeviceThenConnectsDevice() {
```
Comment (CrisBarreiro) [inline sync/sync-impl/src/test/java/com/duckduckgo/sync/impl/AppSyncAccountRepositoryTest.kt:362]: nit: Rename to change references from connect to process code

File: sync/sync-impl/src/test/java/com/duckduckgo/sync/impl/AppSyncAccountRepositoryTest.kt [hunk 2]
```diff
@@ -359,19 +359,19 @@ class AppSyncAccountRepositoryTest {
...
-        val result = syncRepo.connectDevice(jsonConnectKeyEncoded)
+        val result = syncRepo.processCode(jsonConnectKeyEncoded)
 
         verify(syncApi).connect(token, deviceId, encryptedRecoveryCode)
         assertTrue(result is Success)
     }
 
     @Test
-    fun whenConnectingUsingQRFromUnauthenticatedDeviceThenAccountCreatedAndConnects() {
+    fun whenConnectingFromUnauthenticatedDeviceThenAccountCreatedAndConnects() {
```
Comment (CrisBarreiro) [inline sync/sync-impl/src/test/java/com/duckduckgo/sync/impl/AppSyncAccountRepositoryTest.kt:374]: Same here

File: sync/sync-impl/src/test/java/com/duckduckgo/sync/impl/ui/EnterCodeViewModelTest.kt [hunk 3]
```diff
@@ -68,17 +67,17 @@ internal class EnterCodeViewModelTest {
...
+        testee.onPasteCodeClicked()
 
         verify(clipboard).pasteFromClipboard()
     }
 
     @Test
     fun whenUserClicksOnPasteCodeWithRecoveryCodeThenLoginWithCode() = runTest {
         whenever(clipboard.pasteFromClipboard()).thenReturn(jsonRecoveryKeyEncoded)
-        whenever(syncAccountRepository.login(jsonRecoveryKeyEncoded)).thenReturn(Success(true))
+        whenever(syncAccountRepository.processCode(jsonRecoveryKeyEncoded)).thenReturn(Success(true))
```
Comment (CrisBarreiro) [inline sync/sync-impl/src/test/java/com/duckduckgo/sync/impl/ui/EnterCodeViewModelTest.kt:78]: Same for the tests in this file

File: sync/sync-impl/src/test/java/com/duckduckgo/sync/impl/ui/SyncLoginViewModelTest.kt [hunk 4]
```diff
@@ -58,10 +58,10 @@ class SyncLoginViewModelTest {
 
     @Test
     fun whenQRScannedThenPerformLoginAndEmitResult() = runTest {
-        whenever(syncRepostitory.login(jsonRecoveryKeyEncoded)).thenReturn(Success(true))
+        whenever(syncRepostitory.processCode(jsonRecoveryKeyEncoded)).thenReturn(Success(true))
```
Comment (CrisBarreiro) [inline sync/sync-impl/src/test/java/com/duckduckgo/sync/impl/ui/SyncLoginViewModelTest.kt:61]: Same here

**Review by CrisBarreiro — APPROVED** [review]
LGTM!

**Pull request merged**